File: javascript/models/review.js
var Review = function createReview() {
  var id = 0;

  return class Review {
    constructor(customer_id, restaurant_id) {
      this.id = ++id;
      this.customer_id = customer_id
      this.restaurant_id = restaurant_id

      store.reviews = [...store.reviews, this]
    }

    static all() { 
      return store.reviews
    }
  }
}()

Review.prototype.customer = () => {
  return store.customers[this.customer_id-1]
}

Review.prototype.restaurant = () => {
  return store.restaurants[this.restaurant_id-1]
}

File: object_relations/app/models/review.rb
class Review
  attr_accessor :review_text
  @@all = []

  def initialize(review_text, customer, restaurant)
    @review_text = review_text
    @customer = customer
    @restaurant = restaurant
    @restaurant.customer_list << @customer
    @restaurant.review_list << self
    @@all << self
  end

  def customer
    @customer
  end

  def restaurant
    @restaurant
  end

  def self.all
    @@all
  end

end
